src/kvstore.c

```diff
@@ -729,11 +733,15 @@ unsigned int kvstoreHashtableSampleEntries(kvstore *kvs, int didx, void **dst, u
 }
 
 bool kvstoreHashtableExpand(kvstore *kvs, int didx, unsigned long size) {
-    hashtable *ht = kvstoreGetHashtable(kvs, didx);
-    if (!ht) return false;
+    hashtable *ht = createHashtableIfNeeded(kvs, didx);
```

Comment:
Shall we skip it if `size == 0`?

```suggestion
    if (size == 0) return false;
    hashtable *ht = createHashtableIfNeeded(kvs, didx);
```

Reply:
Do you think we need to judge internally or externally? I am worried about its return value, such as the following code, we return true when newsize == 0, if kvstoreHashtableTryExpand return false when newsize is 0, they're both a bit strange to me. i think i will add they both (internally and externally)
```
bool kvstoreExpand(kvstore *kvs, uint64_t newsize, int try_expand, kvstoreExpandShouldSkipHashtableIndex *skip_cb) {
    if (newsize == 0) return true;
    for (int i = 0; i < kvs->num_hashtables; i++) {
        if (skip_cb && skip_cb(i)) continue;
        if (try_expand) {
            if (!kvstoreHashtableTryExpand(kvs, i, newsize)) return false;
        } else {
            kvstoreHashtableExpand(kvs, i, newsize);
        }
    }

    return true;
}
```

Reply:
Good point. Only externally is OK too.

Reply:
Is this incorrect API usage altogether? Should we crash if the expand request is for `0` ?